src/linear_algebra/cross_product.lean

```diff
@@ -0,0 +1,102 @@
+/-
+Copyright (c) 2021 Martin Dvorak. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Martin Dvorak, Kyle Miller, Eric Wieser
+-/
+
+import linear_algebra.bilinear_map
+import linear_algebra.matrix.determinant
+import linear_algebra.vectors
+
+/-!
+# Cross products
+
+This module defines the cross product of vectors in $R^3$ for $R$ a commutative ring.
+
+## Main definitions
+
+* `cross_product` is the cross product of pairs of vectors in $R^3$.
+
+## Main results
+
+* `triple_product_eq_det`
+
+## Notation
+
+The locale `vectors` gives the following notation:
+
+* `×₃` for the cross product
+
+## Tags
+
+crossproduct
+-/
+
+open_locale vectors
+
+variables {R : Type*} [comm_ring R]
+
+/-- The cross product of two vectors in $R^3$ for $R$ a commutative ring. -/
+def cross_product : (fin 3 → R) →ₗ[R] (fin 3 → R) →ₗ[R] (fin 3 → R) :=
+begin
+  apply linear_map.mk₂ R (λ (a b : fin 3 → R),
+    ![ (a 1)*(b 2) - (a 2)*(b 1) ,
+       (a 2)*(b 0) - (a 0)*(b 2) ,
+       (a 0)*(b 1) - (a 1)*(b 0) ]);
+  intros;
+  simp only [vec3_add,
+    pi.add_apply, algebra.id.smul_eq_mul, matrix.smul_cons, matrix.smul_empty, pi.smul_apply];
+  apply vec3_eq;
+  ring,
+end
+
+localized "infixl ` ×₃ `: 68 := cross_product" in vectors
+
+lemma cross_product_def (a b : fin 3 → R) :
+  a ×₃ b =
+    ![ (a 1)*(b 2) - (a 2)*(b 1) ,
+       (a 2)*(b 0) - (a 0)*(b 2) ,
+       (a 0)*(b 1) - (a 1)*(b 0) ] :=
+rfl
+
+lemma cross_product_anticomm (v w : fin 3 → R) :
+  - (v ×₃ w) = w ×₃ v :=
+by simp [cross_product_def, mul_comm]
+
+lemma cross_product_anticomm' (v w : fin 3 → R) :
+  v ×₃ w + w ×₃ v = 0 :=
+by rw [add_eq_zero_iff_eq_neg, cross_product_anticomm]
+
+lemma cross_product_self_eq_zero_vector (v : fin 3 → R) :
+  v ×₃ v = 0 :=
+by simp [cross_product_def, mul_comm]
+
+/-- The cross product of two vectors is perpendicular to the first vector. -/
+lemma dot_self_cross_product_eq_zero (v w : fin 3 → R) :
+  v ⬝ (v ×₃ w) = 0 :=
+by simp [cross_product_def, vec3_dot_product, mul_sub, mul_assoc, mul_left_comm]
+
+/-- The cross product of two vectors is perpendicular to the second vector. -/
+lemma dot_cross_product_self_eq_zero (v w : fin 3 → R) :
+  w ⬝ (v ×₃ w) = 0 :=
+by rw [← cross_product_anticomm, matrix.dot_product_neg, dot_self_cross_product_eq_zero, neg_zero]
```

Comment:
Similarly the `eq_zero` can be dropped here, just as we call the lemma `mul_zero` and not `mul_zero_eq_zero`.
```suggestion
/-- The cross product of two vectors is perpendicular to the first vector. -/
lemma dot_self_cross_product (v w : fin 3 → R) : v ⬝ (v ×₃ w) = 0 :=
by simp [cross_product_def, vec3_dot_product, mul_sub, mul_assoc, mul_left_comm]

/-- The cross product of two vectors is perpendicular to the second vector. -/
lemma dot_cross_product_self (v w : fin 3 → R) : w ⬝ (v ×₃ w) = 0 :=
by rw [← cross_product_anticomm, matrix.dot_product_neg, dot_self_cross_product_eq_zero, neg_zero]
```

Reply:
Is it a general convention for all theorems that have 0 on the right side?

Reply:
Usually when the RHS is obvious, we omit it from the name of the lemma to keep the lemma name short.

Reply:
It is less obvious than `mul_zero` but if this convention is used everywhere there is 0 on the right side, I will change it, of course.

Reply:
I am not sure what thumb up means. Is it within the scope of the mathlib's convention?

Reply:
I guess so. I will remove all the `eq_zero` thingies.